src/dashboard/infrastructure/connection/site-kit-is-connected-call.php

```diff
@@ -0,0 +1,49 @@
+<?php
+// phpcs:disable Yoast.NamingConventions.NamespaceName.TooLong
+// phpcs:disable Yoast.NamingConventions.NamespaceName.MaxExceeded
+namespace Yoast\WP\SEO\Dashboard\Infrastructure\Connection;
+
+use WP_REST_Request;
+
+/**
+ * Class that hold the code to do the REST call to the Site Kit api.
+ *
+ * @phpcs:disable Yoast.NamingConventions.ObjectNameDepth.MaxExceeded
+ */
+class Site_Kit_Is_Connected_Call {
```

Comment:
We need to consider adding micro-cache around both class functions because on dashboard page load, there will be repeated calls to SK's REST endpoints, for script data and for our API calls ( via the `is_onboarded()` function).

However, let's leave that for the caching task, because it needs careful planning (I added a note [there](https://github.com/Yoast/roadmap/issues/558)).

Reply:
The way this is now implemented there are no actual API calls being done. Its basically just calling a function. So I am not sure if caching would be worth the effort here. we can do some performance tests.